protocol-designer/src/components/StepEditForm/index.tsx

```diff
@@ -124,6 +137,12 @@ const StepEditFormManager = (
     formData,
     handleChangeFormInput
   )
+  let handleSave
+  if (isPristineSetTempForm === true) {
+    handleSave = confirmAddPauseUntilTempStep
+  } else if (isPristineSetHeaterShakerTempForm === true) {
+    handleSave = confirmAddPauseUntilHeaterShakerTempStep
+  } else handleSave = saveStepForm
```

Comment:
while this works, `handleSave` gets `any` typed which isn't great.

to avoid this, you can just instantiate `handleSave` as `saveStepForm`, and then conditionally overwrite it in `if` statements!
```suggestion
let handleSave = saveStepForm
  if (isPristineSetTempForm === true) {
    handleSave = confirmAddPauseUntilTempStep
  } else if (isPristineSetHeaterShakerTempForm === true) {
    handleSave = confirmAddPauseUntilHeaterShakerTempStep
  }
```